tsdb/index/inmem/meta.go

```diff
@@ -395,6 +395,20 @@ func (m *Measurement) TagSets(shardID uint64, opt influxql.IteratorOptions) ([]*
 	return sortedTagsSets, nil
 }
 
+func (m *Measurement) AuthorizeRead(auth influxql.Authorizer) bool {
```

Comment:
Currently the only primitive exposed from the `Authorizer` is read/write access to a specific series. Ideally, yes, there would be a way to invert the rules to more efficiently filter through available series. @jwilder and I discussed possibly trying to extract an `influxql.Expr` from the rules, but they might have to be on a per-measurement basis (potentially large tree checking `_name` for different values). Also, since the representation of a measurement and the relevant series is per-index, we would have to try to export primitives useful to both `inmem` and `tsi`, and `MeasurementNames` already takes an `Expr` for filtering.

I'm all for finding a better way, which is why I asked for some early feedback. This works with what we have now, and is a starting point for improvements. :-)